### PR TITLE
LP-1943: Job board list page unit tests for all filter scenarios

### PR DESCRIPTION
**Pull request by mabdullahz**
### Description

[LP-1943](https://philanthropyu.atlassian.net/browse/LP-1943)

### CODE REVIEW

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 1]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_FREELANCE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_freelance_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_FREELANCE_KEY)
+
+    def test_job_list_view_filters_job_location(self):
+        job = JobFactory(country='PK', city='Karachi')
+        response = self.client.get(reverse(
+            'job_list') + '?' + JOB_PARAM_CITY_KEY + '=' + job.city + '&' + JOB_PARAM_COUNTRY_KEY + '=' + job.country.name)
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply openedx/features/job_board/tests/views/test_job_board.py:207]: I have formatted the code using PyCharm’s internal formatter. This is the best that can be done for line 206-207 without breaking functionality.

[the remark this answers is not in the document]
Comment (saad-waqar) [reply openedx/features/job_board/tests/views/test_job_board.py:207]: @mabdullahz 
```
from w3lib.url import add_or_replace_parameters

params = {JOB_PARAM_CITY_KEY: job.city, JOB_PARAM_COUNTRY_KEY: job.country.name}
response = self.client.get(add_or_replace_parameters(reverse('job_list'), params))
```

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 2]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.type, JOB_TYPE_REMOTE_KEY)
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_TYPE_ONSITE_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_TYPE_ONSITE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_onsite_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.type, JOB_TYPE_ONSITE_KEY)
```
Comment (zahid-arbisoft) [inline openedx/features/job_board/tests/views/test_job_board.py:128]: Same here

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 3]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+        number_of_hourly_jobs = 0
+        number_of_salaried_jobs = 0
+
+        for job in self.jobs:
+            if job.compensation is JOB_COMP_VOLUNTEER_KEY:
+                number_of_volunteer_jobs += 1
+            if job.compensation is JOB_COMP_HOURLY_KEY:
+                number_of_hourly_jobs += 1
+            if job.compensation is JOB_COMP_SALARIED_KEY:
+                number_of_salaried_jobs += 1
```
Comment (zahid-arbisoft) [inline openedx/features/job_board/tests/views/test_job_board.py:141]: Lets use Counter here as well

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 4]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_COMP_VOLUNTEER_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_COMP_VOLUNTEER_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_volunteer_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.compensation, JOB_COMP_VOLUNTEER_KEY)
```
Comment (zahid-arbisoft) [inline openedx/features/job_board/tests/views/test_job_board.py:149]: Please review this test as well, same issue. Technically test passed but this `assertEqual` never called neither called for other types of `job.compensation` below

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 5]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_FREELANCE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_freelance_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_FREELANCE_KEY)
+
+    def test_job_list_view_filters_job_location(self):
+        job = JobFactory(country='PK', city='Karachi')
+        response = self.client.get(reverse(
```
Comment (zahid-arbisoft) [inline openedx/features/job_board/tests/views/test_job_board.py:206]: Lets add comment that job search filter was applied on what type of data, meaning 11 jobs (10 created in setup)

Please do the same for `test_job_list_view_filters_job_query`

File: openedx/features/job_board/tests/views/test_job_board.py [hunk 6]
```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
...
+                number_of_freelance_jobs += 1
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_HOURS_FULLTIME_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_FULLTIME_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_fulltime_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_FULLTIME_KEY)
+
```
Comment (zahid-arbisoft) [inline openedx/features/job_board/tests/views/test_job_board.py:187]: Lets update this test as well, it is similar to above two tests.

**Pull request merged**